File: sheren_to/week_01/day_05/mta/mta/mta.js
console.log("MTA");

const lineL = ["8th", "6th", "Union Square", "3rd", "1st"];
const lineN = ["Times Square", "34th", "28th", "23rd", "Union Square", "8th"];
const line6 = ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"];

let stationsToTravel = [];//array for stations to travel

const interchange = "Union Square";

const getLineByName = function(lineName){
  switch (lineName) {
    case "lineL":
     line = lineL
      break;
      case "lineN":
      line = lineN
        break;
    default:
      line = line6;
  }
  return line;
}
const getDirection = function (stationIndex, destinationIndex) {
    let direction = 0;
    if (stationIndex > destinationIndex) {
      direction = -1;
    } else if (stationIndex < destinationIndex) {
      direction = 1;
    }
    return direction;
}
const appendStationsForLeg = function(startIndexForLeg, destinationIndexForLeg, line, directionLeg){
  console.log("direction:" + directionLeg);
  i = startIndexForLeg;
  while(true){
    if((directionLeg > 0 && i > destinationIndexForLeg) || (directionLeg < 0 && i < destinationIndexForLeg))
      break;

    console.log("push i:" + i)
    stationsToTravel.push(line[i]);
    //console.log("station:" + line[i] + " index:" + i);
    i += directionLeg;
    //debugger;
  }

}
const planTrip = function(startLineName, startStationName, destinationLineName, destinationStationName){
   startLine = getLineByName(startLineName);
   destinationLine = getLineByName(destinationLineName);
   //debugger;
   console.log("start station:" + startStationName);
   startStationIndex = startLine.indexOf(startStationName);
   console.log("start station index:" + startStationIndex);
   console.log("destination:" + destinationStationName);


   let startLineInterchangeIndex = startLine.indexOf(interchange);
   let destinationLineInterchangeIndex = destinationLine.indexOf(interchange);

   let isSameLine = startLineName === destinationLineName;
   if (!isSameLine) {
     destinationStationIndex = destinationLine.indexOf(destinationStationName);
     console.log("destination station index:" + destinationStationIndex);

     console.log("Your stations are on different lines and you will need to change trains at Union Square");
     let directionLeg = getDirection(startStationIndex, startLineInterchangeIndex);
     appendStationsForLeg(startStationIndex, startLineInterchangeIndex, startLine, directionLeg);
     directionLeg = getDirection(destinationLineInterchangeIndex, destinationStationIndex);
     appendStationsForLeg(destinationLineInterchangeIndex + directionLeg, destinationStationIndex, destinationLine, directionLeg);

   } else {
     console.log("Your stations are on the same line.");
     destinationStationIndex = startLine.indexOf(destinationStationName);
     console.log("destination station index:" + destinationStationIndex);
     appendStationsForLeg(startStationIndex, destinationStationIndex, startLine);
   }
   console.log(stationsToTravel);
   for ( let i = 0; i < stationsToTravel.length; i++) {
     console.log( stationsToTravel[i] );
   }
   console.log( stationsToTravel.length + " stops in total.");
}

planTrip("lineL", "8th", "lineN", "Times Square");

// const trainLines = function () {//find a way to see if elements are in the same array, need to differntiate same stations names
//   if ( lineL.includes("8th") && lineL.includes("Union Square") ) {
//     console.log('Stattions are on the same line.')
//   } else {
//     console.log('Stations are on different lines and you will need to change trains at Union Square.')
//   }
// };
//
// //if trains are on the same line
// const sameLine = function ( stops ) {
//   let stationStops = []; // new array
//   for ( let i = a; i < b + 1; i++) { // need to change line for each line travelled
//     //let i index = a? a should be a number
//      console.log( stops );//loop to find stations from a to b
//     //list stations in array
//     //use new array to sum total number of stations
//   }
//   stationStops.push( stops );
//   console.log( stationStops );
// };
//

// //if trains are not on the same line
// const diffLine = function () {
//   for ( let i =  ; i < lineL.length - 1; i++) { // need to change from one line to the next
//     //let i index = a? a should be a number
//     //loop to find stations from a to c then c to b
//     //list stations in array from a to c, then push same stations to same array from c to b
//     //use new array to sum total number of stations
//   }
// };
//
// //push stations into new array of stations from a to b or a to c then c to b
// //newarray.length -1 = number of stations

File: sheren_to/week_04/day_02/mta.rb
#mta
require 'pry'

# N = %w( Times Square, 34th, 28th, 23rd, Union Square, 8th )
# L = %w( 8th, 6th, Union Square, 3rd, 1st )
# 6 = %w( Grand Central, 33rd, 28th, 23rd, Union Square, Astor Place )

line_N = [ "Times Square", "34th", "28th", "23rd", "Union Square", "8th" ]
line_L = [ "8th", "6th", "Union Square", "3rd", "1st" ]
line_6 = [ "Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place" ]

#Note: same name station: 8th on line N and L, 28th for both N and 6, 23rd for both N and 6

#Try N-Line from Times Square to Union Square
start = "Times Square"
destination = "Union Square"

#what is start station and destinantion station
#search to see on which line start is, destination is
#search to see if both stations are on same line, if not to Union Square then to station

start_index_N = line_N.index(start).to_i
puts "The index of #{start} in array N is #{start_index_N}"

destination_index_N = line_N.index(destination).to_i
puts "The index of #{destination} in array N is #{destination_index_N}"

stations_to_travel = line_N[start_index_N..destination_index_N]
puts "Stations to travel: #{stations_to_travel}"

puts "Number of stations to travel: #{stations_to_travel.count}"


#def startLine
#def destination_line
# start_station 'Time Square', 'Union Square'
def start_station(start, destination)

 #start_line = prompt("Please select station travelling from: ").downcase

   case start_line

    when line_N.include?(start)
       puts "The starting station is on line N"
    when line_L.include?(start)

      puts "The starting station is on line L"
    when line_6.includes?(start)

      puts "The starting station is on line 6"
    end
  end

def destination_station(start, destination)

  #destination_line = prompt("Please select station travelling to: ").downcase

  case destination_line
    when line_N.include?(destination)

      puts "The destination station is on line N"
    when line_L.include?(destination)

      puts "The destination station is on line L"
    when line_6.includes?(destination)

      puts "The destination station is on line 6"
    end
  end

  start_station("Times Square", "Union Square")
  destination_station("Times Square", "Union Square")













  #
